extensions/wikia/Recirculation/RecirculationController.class.php

```diff
@@ -58,9 +58,30 @@ public function footer() {
 		$this->response->setVal( 'communityHeaderBackground',
 			$themeSettings->getCommunityHeaderBackgroundUrl() );
 		$this->response->setVal( 'sitename', $wgSitename );
+		$this->response->setVal( 'topWikiArticles', $this->getTopWikiArticles() );
+		// TODO if topWikiArticles is empty show one more N&S article card instead of More from wiki card
 		$this->response->setTemplateEngine( WikiaResponse::TEMPLATE_ENGINE_PHP );
 	}
 
+	private function getTopWikiArticles() {
+		global $wgCityId;
+
+		$topWikiArticles = TopWikiArticles::getArticlesWithCache( $wgCityId, false );
+		// do not show itself
+		$topWikiArticles = array_filter( $topWikiArticles, function ( $item ) {
+			return $item['id'] !== RequestContext::getMain()->getTitle()->getArticleID();
+		} );
+		// add index to items to render it by mustache template
+		$topWikiArticles = array_map( function ( $item, $index ) {
+			$item['index'] = $index + 1;
+
+			return $item;
+		}, $topWikiArticles, array_keys( array_values( $topWikiArticles ) ) );
+
+		// show max 3 elements
+		return array_slice( $topWikiArticles, 0, 3 );
```

Comment:
I'd change the order of operations. Slice as soon as possible and iterate on the final set.